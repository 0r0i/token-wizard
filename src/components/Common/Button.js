import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
