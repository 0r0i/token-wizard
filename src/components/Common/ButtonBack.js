import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
