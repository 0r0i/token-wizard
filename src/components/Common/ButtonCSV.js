import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
