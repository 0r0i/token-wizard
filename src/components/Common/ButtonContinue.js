import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
