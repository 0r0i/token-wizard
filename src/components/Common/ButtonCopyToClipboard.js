import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
