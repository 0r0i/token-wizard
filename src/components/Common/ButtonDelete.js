import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
