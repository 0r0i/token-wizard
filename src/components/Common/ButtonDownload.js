import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
