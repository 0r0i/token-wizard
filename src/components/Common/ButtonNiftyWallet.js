import React from 'react'

export const ButtonNiftyWallet = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`btn-ButtonNiftyWallet ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    <span className="btn-ButtonNiftyWallet_Icon" />
    <span className="btn-ButtonNiftyWallet_TextContainer">
      <span className="btn-ButtonNiftyWallet_TopText">Download</span>
      <span className="btn-ButtonNiftyWallet_BottomText">Nifty Wallet</span>
    </span>
  </button>
)
