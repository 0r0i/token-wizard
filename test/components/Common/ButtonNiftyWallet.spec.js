import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { ButtonNiftyWallet } from '../../../src/components/Common/ButtonNiftyWallet'

configure({ adapter: new Adapter() })

describe(`ButtonNiftyWallet`, () => {
  it(`should be a <button> HTML tag`, () => {
    const wrapper = shallow(<ButtonNiftyWallet />)

    expect(wrapper.type()).toBe('button')
  })

  it(`its type attribute should be 'button'`, () => {
    const wrapper = shallow(<ButtonNiftyWallet />)

    expect(wrapper.props().type).toBe('button')
  })

  it(`should render ButtonNiftyWallet component`, () => {
    // Given
    const component = render.create(<ButtonNiftyWallet onClick={jest.fn()} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
