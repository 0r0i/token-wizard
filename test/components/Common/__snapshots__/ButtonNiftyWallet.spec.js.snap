// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNiftyWallet should render ButtonNiftyWallet component 1`] = `
<button
  className="btn-ButtonNiftyWallet "
  onClick={[MockFunction]}
  type="button"
>
  <span
    className="btn-ButtonNiftyWallet_Icon"
  />
  <span
    className="btn-ButtonNiftyWallet_TextContainer"
  >
    <span
      className="btn-ButtonNiftyWallet_TopText"
    >
      Download
    </span>
    <span
      className="btn-ButtonNiftyWallet_BottomText"
    >
      Nifty Wallet
    </span>
  </span>
</button>
`;
